src/diagonal_transition.rs

```diff
@@ -0,0 +1,2434 @@
+use std::mem::swap;
```

Comment:
Move to a subdirectory `src/others/` (or something else if you find a better name)
Same for `ukkonen.rs`